src/app/providers/electron.service.ts

```diff
@@ -38,16 +38,23 @@ export class ElectronService {
 
   public activateBoard(board:Board){
       var windows = board.windows;
-      var scriptContent = `@ECHO OFF
-      `;
-
-      for (let window of windows){
-        scriptContent += this.createScriptCommand(window);
+      var scriptCommand = "";
+      var index = 0;
+      function activateWindow(window: Window, electronService: ElectronService){
+        setTimeout(function(){
+          if(window.windowFilePath != null && window.windowFilePath != ""){
+            scriptCommand = electronService.createScriptCommand(window);
+            console.log(scriptCommand);
+            console.log(electronService.scriptFilePath);
+            electronService.executeScript(scriptCommand);
+          }
+          index++;
+          if(index < windows.length){
+            activateWindow(windows[index], electronService);
+          }
+        }, 3000)
```

Comment:
I see what you're trying to do here, but I think it would have been easier (and also cleaner code wise) to just add a "wait" command to the beginning of the .bat file.